packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
+        populatedNestedTemplates.deployedNestedTemplate,
+        nestedStackTemplates.nestedStackName,
+        sdk,
+      );
+    }
+  }
+
+  private async getNestedStackTemplates(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
+    parentStackName: string | undefined, listStackResources: ListStackResources | undefined, sdk: ISDK,
+  ): Promise<NestedStackTemplates> {
+    const nestedTemplatePath = path.join(rootStackArtifact.assembly.directory, nestedTemplateAssetPath);
+    // CFN generates the nested stack name in the form `ParentStackName-NestedStackLogicalID-SomeHashWeCan'tCompute,
+    // so we get the ARN and manually extract the name.
+    //const nestedStackArn = (await listStackResources?.listStackResources())?.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+    let nestedStackArn: string | undefined = undefined;
+
+    try {
+      const stackResources = await listStackResources?.listStackResources();
+
+      if (stackResources) {
+        nestedStackArn = stackResources.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+      }
+    } catch (e) {
+      if (e.message !== `Stack with id ${parentStackName} does not exist`) {
+        throw e;
+      }
+    }
+    const nestedStackName = nestedStackArn?.slice(nestedStackArn.indexOf('/') + 1, nestedStackArn.lastIndexOf('/'));
+
+    return {
+      generatedNestedTemplate: JSON.parse(fs.readFileSync(nestedTemplatePath, 'utf-8')),
+      deployedNestedTemplate: nestedStackName
+        ? await this.readCurrentStackTemplate(rootStackArtifact, nestedStackName, sdk) : { Resources: {} },
+      nestedStackName,
+    };
+  }
+
+  private isCdkManagedNestedStack(stackResource: any): stackResource is NestedStackResource {
+    return stackResource.Type === 'AWS::CloudFormation::Stack' && stackResource.Metadata && stackResource.Metadata['aws:asset:path'];
+  }
+
+  private populateParentTemplates(
+    generatedNestedTemplate: any,
+    deployedNestedTemplate: any = {},
+    nestedStackTemplates: NestedStackTemplates,
+  ): NestedStackTemplates {
+    if (!generatedNestedTemplate.Properties) {
+      generatedNestedTemplate.Properties = {};
+    }
+
+    generatedNestedTemplate.Properties.NestedTemplate = nestedStackTemplates.generatedNestedTemplate;
+    const generatedNestedTemplateProperty = generatedNestedTemplate.Properties.NestedTemplate;
+
+    if (!deployedNestedTemplate.Properties) {
+      deployedNestedTemplate.Properties = {};
+    }
+
+    deployedNestedTemplate.Properties.NestedTemplate = nestedStackTemplates.deployedNestedTemplate;
+    const deployedNestedTemplateProperty = deployedNestedTemplate.Properties.NestedTemplate ?? {};
+
+    return {
+      generatedNestedTemplate: generatedNestedTemplateProperty,
+      deployedNestedTemplate: deployedNestedTemplateProperty,
+      nestedStackName: '',
```

Comment:
How can passing `''` here be correct...?

Reply:
I'll admit this is very confusing. This is just reusing the nested stack template interface from before because this method returns a generated and deployed template, but it's not actually doing a lookup so the stack name isn't needed (and is passed as empty here).